### PR TITLE
Category crud endpoints

### PR DESCRIPTION
**Pull request by NatMyronenko**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -55,12 +55,6 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-test</artifactId>
             <scope>test</scope>
-            <exclusions>
```
Comment (alex97po) [inline pom.xml:58]: Please, please, revert ANY changes from pom.xml

File: src/main/java/com/example/java/test/junior/developer/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import lombok.RequiredArgsConstructor;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import java.util.List;
+
+@RestController
+@RequiredArgsConstructor
+@RequestMapping("/api/v1/users")
+public class UserController {
```
Comment (alex97po) [inline src/main/java/com/example/java/test/junior/developer/controller/UserController.java:14]: Why we have changes for user in this PR?

**Review by alex97po — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/java/test/junior/developer/model/Language.java [hunk 3]
```diff
@@ -26,5 +25,11 @@ public class Language {
     @Column(name = "language", nullable = false)
     public String name;
 
+    @OneToMany(cascade = CascadeType.ALL,
+            mappedBy = "language",
+            fetch = FetchType.EAGER)
+    private List<Category> list_categories;
```
Comment (alex97po) [inline src/main/java/com/example/java/test/junior/developer/model/Language.java:31]: Java Code Convention violation - NEVER use underscore in names in Java, only camel case.

File: src/main/resources/application.properties [hunk 4]
```diff
@@ -1,10 +1,10 @@
 #for local db (swagger,postman etc)
 
-spring.datasource.url=jdbc:h2:mem:demo;DB_CLOSE_ON_EXIT=FALSE
-spring.datasource.username=sa
-spring.datasource.password=
-spring.datasource.driver-class-name=org.h2.Driver
-spring.jpa.database-platform=org.hibernate.dialect.H2Dialect
+#spring.datasource.url=jdbc:h2:mem:demo;DB_CLOSE_ON_EXIT=FALSE
```
Comment (alex97po) [inline src/main/resources/application.properties:3]: Please, revert ANY changes in application.properties

**Review by alex97po — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 4 above)
Comment (NatMyronenko) [reply src/main/resources/application.properties:3]: i dont know how its possible, but i dont have this red part of code in my prop file

Comment (NatMyronenko) [reply src/main/resources/application.properties:3]: i deleted red part of code and lieve only green line. but i cheked and i see, master has this part of code

File: src/test/java/com/example/java/test/junior/developer/category/CategoryServiceJunitTest.java [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class CategoryServiceJunitTest {
+    private final CategoryRepository categoryRepository = Mockito.mock(CategoryRepository.class);
+    private final CategoryMapper categoryMapper = new CategoryMapper();
```
Comment (alex97po) [inline src/test/java/com/example/java/test/junior/developer/category/CategoryServiceJunitTest.java:21]: The idea of unit test is that all dependency components are mocked. So CategoryMapper should be mocked too.

**Review by alex97po — CHANGES_REQUESTED** [review]

**Review by alex97po — APPROVED** [review]

**Pull request merged**